src/udev/net/ethtool-util.c

```diff
@@ -615,6 +615,8 @@ int ethtool_set_glinksettings(int *fd, const char *ifname, struct link_config *l
         u->base.autoneg = link->autonegotiation;
 
         if (!eqzero(link->advertise)) {
+                if (link->autonegotiation == -1)
+                        u->base.autoneg = 1;
```

Comment:
can you explain this? why is this inside the `!eqzero(link->advertise)` if block, and not before it?

Reply:
I mainly wanted to prevent any accidental behavior changes since I am not in my comfort zone here. Setting this to 1 is necessary, because the default of -1 makes the kernel turn it off altogether. Should I make the assignment outside conditional and the one inside non-conditional? I think that would make it work like ethtool does.

Reply:
well, I don't know the kernel api either there, i.e. what the autoneg field expects. before changing this we should really figure that out...

Reply:
Afaik, it's either AUTONEG_ENABLE or AUTONEG_DISABLE. Anything else gives an EINVAL. So I guess we really should avoid assigning -1.

Reply:
https://github.com/torvalds/linux/blob/b71acb0e372160167bf6d5500b88b30b52ccef6e/drivers/net/phy/phy.c#L273